From b98fd85be77f96a82159f23651e7045711c1ae9e Mon Sep 17 00:00:00 2001
From: Bret Liu <Bret_liu@tw.shuttle.com>
Date: Wed, 25 May 2016 15:27:00 +0800
Subject: [PATCH 081/443] Complete menu page.     1. add permission session
 detect function to menu.html.     2. change menu item icon.     3. delete
 navigation bar link.

---
 login/views.py      |  53 ++++++++-------
 templates/menu.html | 182 ++++++++++++++++++++--------------------------------
 2 files changed, 101 insertions(+), 134 deletions(-)

diff --git a/login/views.py b/login/views.py
index 0ef28f4..f1e501f 100644
--- a/login/views.py
+++ b/login/views.py
@@ -1,7 +1,7 @@
 from django.shortcuts import render
 from django.http import JsonResponse, HttpResponse, HttpResponseRedirect
 from django.template import loader
-from api.models import Users, Systemlogs
+from api.models import Users, Systemlogs, Acl, Role
 from django import forms
 from nocaptcha_recaptcha.fields import NoReCaptchaField
 from api.utility import md5
@@ -20,14 +20,11 @@ key = 15
 def check_user(request):
     mail = request.POST.get('email', '')
     password = md5(request.POST.get('password', ''))
-    #print("mail",mail)
-    #print("password",password)
-    form = RegisterForm(request.POST)
     data = request.POST
     response = data.get('g-recaptcha-response')
     if checkRecaptcha(response,SECRET_KEY):
         try:
-            obj = Users.objects.get(Email=mail, Password=password)
+            obj = Users.objects.get(Email = mail, Password = password)
             # set session
             request.session['Account'] = obj.Account
             request.session['Email'] = obj.Email
@@ -41,10 +38,11 @@ def check_user(request):
         print("reCAPTCHA validation failed")
         return json.dumps(res_code[1])
 
+
 def change_passwd(request):
-    print("change_passwd")
     if not 'OId' in request.session:
         return HttpResponseRedirect("/")
+
     check_result = json.loads(check_user(request))
     if check_result["CODE"] == "0":
         mail = request.session['Email']
@@ -52,9 +50,9 @@ def change_passwd(request):
         password = md5(request.POST.get('password', ''))
         new_password = request.POST.get('new_password', '')
         confirm_password = request.POST.get('confirm_password', '')
-        obj = Users.objects.filter(Email=mail, Password=password)
+        obj = Users.objects.filter(Email = mail, Password = password)
         if len(obj) == 1:
-            obj.update(Password=new_password,Name=name)
+            obj.update(Password=new_password,Name = name)
             return HttpResponse(json.dumps(res_code[0]))
         else:
             return HttpResponse(json.dumps(res_code[1]))
@@ -63,15 +61,25 @@ def change_passwd(request):
     else:
         return HttpResponse(json.dumps(res_code[1]))
 
-class RegisterForm(forms.Form):
-    captcha = NoReCaptchaField()
+
+def setSessionPermission(request):
+    try:
+        request.session['permission_system'] = Acl.objects.get(FunctionName = 'permission_system', RoleOId = Users.objects.get(Email = request.session['Email']).RoleOId).ACL
+        request.session['permission_device'] = Acl.objects.get(FunctionName = 'permission_device', RoleOId = Users.objects.get(Email = request.session['Email']).RoleOId).ACL
+        request.session['permission_fruserlist'] = Acl.objects.get(FunctionName = 'permission_fruserlist', RoleOId = Users.objects.get(Email = request.session['Email']).RoleOId).ACL
+        request.session['permission_frusergroup'] = Acl.objects.get(FunctionName = 'permission_frusergroup', RoleOId = Users.objects.get(Email = request.session['Email']).RoleOId).ACL
+        request.session['permission_log'] = Acl.objects.get(FunctionName = 'permission_log', RoleOId = Users.objects.get(Email = request.session['Email']).RoleOId).ACL
+    except Exception as e:
+        print('session permission error', e)
+
 
 def login(request):
     if request.method == "POST":
         check_result = json.loads(check_user(request))
         if check_result["CODE"] == "0":
             logContent= request.session['Name']+' login success.'
-            Systemlogs.objects.create(Category="User",LogContent=logContent)
+            Systemlogs.objects.create(Category="User", LogContent = logContent)
+            setSessionPermission(request)
             return HttpResponseRedirect("/user_list")
         elif check_result["CODE"] == "1":
             return render(request, 'login.html', {'Result':"Fail",'Code': "9998","Message":"reCAPTCHA validation failed"})
@@ -138,7 +146,7 @@ def password_reset_confirm(request, uidb64=None, times=None, token=None,
         time_limit = starttime + timedelta*60
 
         if time_limit < nowtime:
-            print ("Expired time :", time.strftime('%Y-%m-%d %H:%M:%S+00:00',time.localtime(time_limit)))
+            print ("Expired time :", time.strftime('%Y-%m-%d %H:%M:%S+00:00', time.localtime(time_limit)))
             return render(request, 'registration/not_found.html', {'Result':"Fail",'Code': "0000","Message":"Link expired"})
         elif starttime < update_time_ts:
             print ("Password has been changed, Last change password time :", update_time)
@@ -166,20 +174,20 @@ def ResetPasswordRequestView(request):
             account_encrypt = encrypt(key, associated.Account)
             for account in get_users(associated.Email):
                 c = {
-                        'email': associated.Email,
-                        'domain': request.META['HTTP_HOST'],
-                        'site_name': 'Shuttle BR06',
-                        'expired': timedelta,
-                        'uid': account_encrypt,
-                        'user': associated.Account,
-                        'token': br06_token_generator.user_make_token(account),
-                        'times': time_encrypt,
-                        'protocol': 'http',
+                    'email': associated.Email,
+                    'domain': request.META['HTTP_HOST'],
+                    'site_name': 'Shuttle BR06',
+                    'expired': timedelta,
+                    'uid': account_encrypt,
+                    'user': associated.Account,
+                    'token': br06_token_generator.user_make_token(account),
+                    'times': time_encrypt,
+                    'protocol': 'http',
                 }
             subject_template_name='Password reset on Shuttle BR06'
             email_template_name='registration/password_reset_email.html'
             email = loader.render_to_string(email_template_name, c)
-            send_mail(subject_template_name, email, DEFAULT_FROM_EMAIL , [associated.Email], fail_silently=False)
+            send_mail(subject_template_name, email, DEFAULT_FROM_EMAIL , [associated.Email], fail_silently = False)
             return render(request, 'registration/forget_password.html', {'Result':"Success",'Code': "9999"})
 
         except:
@@ -187,5 +195,6 @@ def ResetPasswordRequestView(request):
 
     return render(request, 'registration/forget_password.html', {'CODE': ""})
 
+
 def login_i18n(request):
     return render(request, 'login_i18n.html')
diff --git a/templates/menu.html b/templates/menu.html
index 9392bb4..9093e0a 100644
--- a/templates/menu.html
+++ b/templates/menu.html
@@ -1,5 +1,6 @@
 {% load i18n %}
 {% load static %}
+
 <link rel="stylesheet" href="{% static "bootstrap.min.css" %}">
 <link rel="stylesheet" href="{% static "datepicker3.css" %}">
 <link rel="stylesheet" href="{% static "bootstrap-table.css" %}">
@@ -11,124 +12,81 @@
 			<button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#sidebar-collapse">
 				<span class="sr-only">Toggle navigation</span>
 				<span class="icon-bar"></span>
-				<span class="icon-bar"></span>
-				<span class="icon-bar"></span>
-			</button>
-			<a class="navbar-brand" href="/home"><span>Shuttle</span>/BR06</a>
-			<ul class="user-menu">
-				<li class="dropdown pull-right">
-					<a href="#" class="dropdown-toggle" data-toggle="dropdown"><svg class="glyph stroked male-user"><use xlink:href="#stroked-male-user"></use></svg> {{ request.session.Name }} <span class="caret"></span></a>
-					<ul class="dropdown-menu" role="menu">
-						<!-- <li><a href="#"><svg class="glyph stroked male-user"><use xlink:href="#stroked-male-user"></use></svg> Profile</a></li> -->
-						<li><a href="/user_setting"><svg class="glyph stroked gear"><use xlink:href="#stroked-gear"></use></svg> {% trans "Settings" %}</a></li>
-						<li><a href="/logout"><svg class="glyph stroked cancel"><use xlink:href="#stroked-cancel"></use></svg> {% trans "Logout" %}</a></li>
-					</ul>
-				</li>
-			</ul>
-		</div>
-
-	</div><!-- /.container-fluid -->
+                <span class="icon-bar"></span>
+                <span class="icon-bar"></span>
+            </button>
+            <a class="navbar-brand"><span>Shuttle</span>/BR06</a>
+            <ul class="user-menu">
+                <li class="dropdown pull-right">
+                    <a href="#" class="dropdown-toggle" data-toggle="dropdown"><svg class="glyph stroked male-user"><use xlink:href="#stroked-male-user"></use></svg> {{ request.session.Name }} </a>
+                </li>
+            </ul>
+        </div>
+    </div>
 </nav>
 
 <div id="sidebar-collapse" class="col-sm-3 col-lg-2 sidebar">
-	<form role="search">
-		<div class="form-group">
-			<!-- <input type="text" class="form-control" placeholder="Search"> -->
-	        </div>
-	</form>
+	<form role="search"><div class="form-group"></div></form>
 	<ul class="nav menu">
-		<li class="parent ">
-			<a href="#">
-			    <span data-toggle="collapse" href="#sub-item-1"><svg class="glyph stroked chevron-down"><use xlink:href="#stroked-chevron-down"></use></svg></span> {% trans "System" %}
-			</a>
+        {% ifequal request.session.permission_system "Enable" %}
+	    <li class="parent ">
+			<a href="#"><span data-toggle="collapse" href="#sub-item-1"><svg class="glyph stroked desktop"><use xlink:href="#stroked-desktop"/></svg> {% trans "System" %} </span></a>
             <ul class="children collapse" id="sub-item-1">
-                <li>
-			    	<a class="" href="/dashboard">
-				    <svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "Dashboard" %}
-					</a>
-				</li>
-			    <li>
-				    <a class="" href="/user_list">
-				    <svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "User List" %}
-				    </a>
-			    </li>
-			        <li>
-				    <a class="" href="/user_group_list">
-				    <svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "Group List" %}
-				    </a>
-			    </li>
-			    <li>
-				    <a class="" href="/role_list">
-				    <svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "Role & Permission" %}
-				    </a>
-			    </li>
-                <li>
-		            <a class="" href="#">
-		            <svg class="glyph stroked gear"><use xlink:href="#stroked-gear"/></svg>  {% trans "Setting" %}
-		            </a>
-		        </li>
-             </ul>
-		</li>
-        <li name="device_list"><a href="/device_list"><svg class="glyph stroked app window with content"><use xlink:href="#stroked-app-window-with-content"/></svg> {% trans "Device List" %}</a></li>
-        <li name="fruser_list"><a href="/fruser_list"><svg class="glyph stroked app window with content"><use xlink:href="#stroked-app-window-with-content"/></svg> {% trans "FRUser List" %}</a></li>
-        <li name="fruser_group"><a href="/fruser_group"><svg class="glyph stroked app window with content"><use xlink:href="#stroked-app-window-with-content"/></svg> {% trans "FRUser Group" %}</a></li>
+                <li><a class="" href="/dashboard"><svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "Dashboard" %} </a></li>
+                <li><a class="" href="/user_list"><svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "User List" %} </a></li>
+                <li><a class="" href="/user_group_list"><svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "Group List" %} </a></li>
+                <li><a class="" href="/role_list"><svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "Role & Permission" %} </a></li>
+                <li><a class="" href="#"><svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"/></svg>{% trans "Setting" %} </a></li>
+            </ul>
+        </li>
+        {% endifequal %}
+        {% ifequal request.session.permission_device "Enable" %}
+        <li name="device_list"><a href="/device_list"><svg class="glyph stroked external hard drive"><use xlink:href="#stroked-external-hard-drive"/></svg> {% trans "Device List" %} </a></li>
+        {% endifequal %}
+        {% ifequal request.session.permission_fruserlist "Enable" %}
+        <li name="fruser_list"><a href="/fruser_list"><svg class="glyph stroked male user "><use xlink:href="#stroked-male-user"/></svg> {% trans "FRUser List" %} </a></li>
+        {% endifequal %}
+        {% ifequal request.session.permission_frusergroup "Enable" %}
+        <li name="fruser_group"><a href="/fruser_group"><svg class="glyph stroked male user "><use xlink:href="#stroked-male-user"/></svg> {% trans "FRUser Group" %} </a></li>
+        {% endifequal %}
+        {% ifequal request.session.permission_log "Enable" %}
         <li class="parent ">
-		    <a href="#">
-			   <span data-toggle="collapse" href="#sub-item-3"><svg class="glyph stroked chevron-down"><use xlink:href="#stroked-chevron-down"></use></svg></span> {% trans "Log" %}
-		    </a>
-		    <ul class="children collapse" id="sub-item-3">
-			<li>
-				<a class="" href="/fruserlog_list">
-				     <svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "FRUser Log List" %}
-				</a>
-			</li>
-            <li>
-				<a class="" href="/systemlog_list">
-				      <svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "System Log List" %}
-     			</a>
-			 </li>
-             <li>
-				<a class="" href="/logsetting">
-				      <svg class="glyph stroked gear"><use xlink:href="#stroked-gear"/></svg> {% trans "Log Setting" %}
-				</a>
-			 </li>
-			 </ul>
-		</li>
-		<li role="presentation" class="divider"></li>
-		<li><a href="/logout"><svg class="glyph stroked cancel"><use xlink:href="#stroked-cancel"/></svg> {% trans "Log Out" %}</a></li>
-	</ul>
-	<script>
-                 $(document).ready(function() {
-                   var systemlink = [ "/user_list", "/user_add/", "/user_group_list", "/role_list", "/user_group_add/","/role_add/","/dashboard/"];
-                   var loglink = ["/fruserlog_list","/systemlog_list","/logsetting"];
-
-                   jQuery.each(systemlink, function() {
-                       if($(location).attr('pathname') ==this){
-                           $('#sub-item-1').collapse('show');
-                       }
-	           });
-
-                   jQuery.each(loglink, function() {
-                       if($(location).attr('pathname') ==this){
-                            $('#sub-item-3').collapse('show');
-                       }
-                  });
-
-                   $('#sub-item-1').collapse({'toggle': false});
-                   $('#sub-item-3').collapse({'toggle': false});
-
-                })
-
-	        $("#sub-item-1").on("shown.bs.collapse", function() {
-			$("#sub-item-3").collapse('hide');
-
-	       })
-
-	        $("#sub-item-3").on("shown.bs.collapse", function() {
-                        $("#sub-item-1").collapse('hide');
+            <a href="#"><span data-toggle="collapse" href="#sub-item-3"><svg class="glyph stroked line-graph"><use xlink:href="#stroked-line-graph"></use></svg> {% trans "Log" %} </span></a>
+            <ul class="children collapse" id="sub-item-3">
+                <li><a class="" href="/fruserlog_list"><svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "FRUser Log List" %} </a></li>
+                <li><a class="" href="/systemlog_list"><svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "System Log List" %} </a></li>
+                <li><a class="" href="/logsetting"><svg class="glyph stroked chevron-right"><use xlink:href="#stroked-chevron-right"></use></svg> {% trans "Log Setting" %} </a></li>
+            </ul>
+        </li>
+        {% endifequal %}
+        <li role="presentation" class="divider"></li>
+        <li><a href="/logout"><svg class="glyph stroked cancel"><use xlink:href="#stroked-cancel"/></svg> {% trans "Log Out" %} </a></li>
+    </ul>
 
-	        })
+    <script>
+        $(document).ready(function() {
+            var systemlink = ["/dashboard/", "/user_list", "/user_add/", "/user_group_list", "/user_group_add", "/role_list", "/role_add/", "/role_permission/", "/user_group_add/"];
+            var loglink = ["/fruserlog_list", "/systemlog_list", "/logsetting"];
+            jQuery.each(systemlink, function() {
+                if($(location).attr('pathname') ==this){
+                    $('#sub-item-1').collapse('show');
+                }
+            });
+            jQuery.each(loglink, function() {
+                if($(location).attr('pathname') ==this){
+                    $('#sub-item-3').collapse('show');
+                }
+            });
+            $('#sub-item-1').collapse({'toggle': false});
+            $('#sub-item-3').collapse({'toggle': false});
+        })
+        $("#sub-item-1").on("shown.bs.collapse", function() {
+            $("#sub-item-3").collapse('hide');
+        })
+        $("#sub-item-3").on("shown.bs.collapse", function() {
+            $("#sub-item-1").collapse('hide');
+        })
+    </script>
 
-	</script>
 	<a class="navbar-fixed-bottom" href="http://MediaLoot.com" style=" color:white"> Theme from MediaLoot</a>
-</div><!--/.sidebar-->
+</div>
-- 
1.9.1

